### ISSUE TITLE
Kubernetes Service can't be exposed if service name is long.

### ISSUE THREAD

**Issue by uthark** [tailscale/tailscale#10807]
### What is the issue?

I have a service name which has 57 characters in the name.
I wanted to expose it but it failed - STS pod had labels longer than 63 characters

```
14m (x22 over 86m)   Warning   FailedCreate        StatefulSet/ts-anim-control-a2x-ag-ax-for-a2x-deployment-ac--s4765   create Pod ts-anim-control-a2x-ag-ax-for-a2x-deployment-ac--s4765-0 in StatefulSet ts-anim-control-a2x-ag-ax-for-a2x-deployment-ac--s4765 failed error: Pod "ts-anim-control-a2x-ag-ax-for-a2x-deployment-ac--s4765-0" is invalid: metadata.labels: Invalid value: "ts-anim-control-a2x-ag-ax-for-a2x-deployment-ac--s4765-6fd556789b": must be no more than 63 characters
```

Workaround is to create a new Service with a shorter name. 

### Steps to reproduce

1. Create a service with a long name.
2. Try to expose it.

### OS

Linux

### OS version

COS / GKE

### Tailscale version

1.56.1

**Comment by irbekrm** [comment]
Hi @uthark could you confirm what operator image tag you are using? I thought we fixed this in https://github.com/tailscale/tailscale/pull/10343

**Comment by uthark** [comment]
image: tailscale/k8s-operator:v1.56.1

**Comment by uthark** [comment]
#10343 is about StatefulSet name, the issue I see is about the labels for the pod that's being created.

Generated sts name is `ts-anim-control-a2x-ag-ax-for-a2x-deployment-ac--fwp2l`

For the pod which kubernetes creates, it adds label `statefulset.kubernetes.io/pod-name` which follows the general limit for the label values.

The issue is that generated pod name is longer than 63 characters.

**Comment by irbekrm** [comment]
It is the same issue (the label size ultimately comes from the `Statefulset` name, we do not control `Pod` names).

I can reproduce it, so there must be a bug, will take a look.

**Comment by irbekrm** [comment]
Hi @uthark thanks again for spotting the issue!

This should hopefully have been fixed in #10825 .`unstable-v1.57.76` image tag/`v1.57.76` Helm chart tag should have the fix. Do let us know if it fixes the issue for you
